src/data/two_pointing.lean

```diff
@@ -0,0 +1,92 @@
+/-
+Copyright (c) 2022 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import data.sum.basic
+import logic.nontrivial
+
+/-!
+# Two-pointings
+
+This file defines `two_pointing α`, the type of two pointings of `α`. A two-pointing is the data of
+two distinct terms.
+
+This is morally a Type-valued `nontrivial`. Another type which is quite close in essence is `sym2`.
+-/
+
+open function
+
+variables {α β : Type*}
+
+/-- Two-pointing of a type. This is a Type-valued termed `nontrivial`. -/
+@[ext, derive decidable_eq] structure two_pointing (α : Type*) :=
+(fst snd : α)
+(fst_ne_snd : fst ≠ snd)
+
+namespace two_pointing
+variables (p : two_pointing α) (q : two_pointing β)
+
+lemma snd_ne_fst : p.snd ≠ p.fst := p.fst_ne_snd.symm
+
+/-- Swaps the two pointed elements. -/
+def swap : two_pointing α := ⟨p.snd, p.fst, p.snd_ne_fst⟩
```

Comment:
Can you put `@[simps]` on this to get two of the following lemmas for free?

Reply:
Actually, doesn't apply anymore.